server/data/data-library/src/main/java/org/apache/james/repository/api/ObjectRepository.java

```diff
@@ -33,5 +35,5 @@
 
     void put(String key, Object value);
 
-    void remove(String key);
+    void remove(String key) throws UsersRepositoryException;

```

Comment:
you should return a boolean on remove, that would make it possible to throw in case of UserRepository without breaking other classes